### ISSUE TITLE
Lynx linter should detect children with no spec

### ISSUE THREAD

**Issue by ansongoldade** [lynx-json/lynx-docs#90]
The following would result in a document that is not a valid application/lynx+json document when the negative section is used because the child would not have a spec with hints on it. This should result in a linter error.

```
parent>container:
  child:
    "#truthy>link"
      href: .
      label>: Go here
    "^truthy": null
```

**Comment by ansongoldade** [comment]
After further discussion, we should consider failing and reporting the issue during processing the template because we can't reliably calculate children until after data binding. Because of that we aren't able to export this template in a way that would work on a dynamic server when the `truthy` value is not truthy.

**Comment by ansongoldade** [comment]
Resolved with PR #100
